examples/model_selection/plot_roc.py

```diff
@@ -1,132 +1,269 @@
+# In a multi-class classification setup with highly imbalanced classes,
+# micro-averaging is preferable over macro-averaging. In such cases, one can
+# alternatively use a weighted macro-averaging, not demoed here.
+
+RocCurveDisplay.from_predictions(
+    y_onehot_test.ravel(),
+    y_score.ravel(),
+    name="One-vs-Rest ROC (micro-averaged)",
+    color="darkorange",
+)
+plt.plot([0, 1], [0, 1], "k--", label="ROC curve for chance level (AUC = 0.5)")
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.0])
+plt.axis("square")
+plt.xlabel("False Positive Rate")
+plt.ylabel("True Positive Rate")
+plt.title("Receiver operating characteristic (micro-averaged)")
+plt.legend()
+plt.show()
+
+# %%
+# In the case where the main interest is not the plot but the ROC-AUC score
+# itself, we can reproduce the value shown in the plot using
+# :class:`~sklearn.metrics.roc_auc_score`.
 
-# First aggregate all false positive rates
-all_fpr = np.unique(np.concatenate([fpr[i] for i in range(n_classes)]))
+from sklearn.metrics import roc_auc_score
+
+micro_roc_auc_ovr = roc_auc_score(
+    y_test,
+    y_score,
+    multi_class="ovr",
+    average="micro",
+)
+
+print(f"Micro-averaged One-vs-Rest ROC AUC score:\n{micro_roc_auc_ovr:.2f}")
+
+# %%
+# This is equivalent to computing the ROC curve with
+# :class:`~sklearn.metrics.roc_curve` and then the area under the curve with
+# :class:`~sklearn.metrics.auc` for the raveled true and predicted classes.
+
+from sklearn.metrics import roc_curve, auc
+
+# store the fpr, tpr, and roc_auc for all averaging strategies
+fpr, tpr, roc_auc = dict(), dict(), dict()
+# Compute micro-average ROC curve and ROC area
+fpr["micro"], tpr["micro"], _ = roc_curve(y_onehot_test.ravel(), y_score.ravel())
+roc_auc["micro"] = auc(fpr["micro"], tpr["micro"])
+
+print(f"Micro-averaged One-vs-Rest ROC AUC score:\n{roc_auc['micro']:.2f}")
+
+# %%
+# .. note:: By default, the computation of the ROC curve adds a single point at
+#     the maximal false positive rate by using linear interpolation and the
+#     McClish correction [:doi:`Analyzing a portion of the ROC curve Med Decis
+#     Making. 1989 Jul-Sep; 9(3):190-5.<10.1177/0272989x8900900307>`].
+#
+# ROC curve using the OvR macro-average
+# -------------------------------------
+#
+# Obtaining the macro-average requires computing the metric independently for
+# each class and then taking the average over them, hence treating all classes
+# equally a priori. We first aggregate the true/false positive rates per class:
 
-# Then interpolate all ROC curves at these points
-mean_tpr = np.zeros_like(all_fpr)
 for i in range(n_classes):
-    mean_tpr += np.interp(all_fpr, fpr[i], tpr[i])
+    fpr[i], tpr[i], _ = roc_curve(y_onehot_test[:, i], y_score[:, i])
+    roc_auc[i] = auc(fpr[i], tpr[i])
 
-# Finally average it and compute AUC
+fpr_grid = np.linspace(0.0, 1.0, 1000)
+
+# Interpolate all ROC curves at these points
+mean_tpr = np.zeros_like(fpr_grid)
+
+for i in range(n_classes):
```

Comment:
If you don't want to use `sp.interpolate.interp1d` then we should add a comment that `interp` is doing linear interpolation. Another downside of using `interp` is that values need to be ordered.

Reply:
I am hesitating about what is the best, as we actually use `np.interp` to compute the tpr in the private function `_binary_roc_auc_score`. Also it would add boilerplate code below. Maybe the easiest is to indeed add a comment on linear interpolation.